src/elements/e-video.vue

```diff
@@ -0,0 +1,254 @@
+       *
+       * Default from iframe is 2
+       */
+      frameborder: {
+        type: [String, Number],
+        default: 2,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       */
+      width: {
+        type: [String, Number],
+        default: 560,
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+      /**
+       * Defines the players size to be responsive 100% width to the parent container
+       */
+      responsive: {
+        type: Boolean,
+        default: false,
+      },
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = playerUrlSources[this.source];
+
+        if (!(this.videoId || this.videoUrl)) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
+        }
+
+        if (this.videoUrl && !this.videoId) {
+          if (!this.validateUrl()) {
+            return undefined;
+          }
+
+          if (this.source === 'youtube') {
+            /**
+             * Extract id from video url
+             */
+            const id = this.videoUrl.substring(
+              this.videoUrl.indexOf('v=') + 2,
+              this.videoUrl.indexOf('&')
+            );
+
+            return `${playerBaseUrl}${id}`;
+          }
+
+          const id = this.videoUrl.substring(
+            this.videoUrl.indexOf(videoUrlSources[this.source])
+              + videoUrlSources[this.source].length,
+            this.videoUrl.length
+          );
+
+          return `${playerBaseUrl}${id}`;
+        }
+
+        if (this.videoId.includes(videoUrlSources[this.source])) {
+          // eslint-disable-next-line no-console
+          console.error('video-id can not be part of a URL');
+
+          return undefined;
+        }
+
+        return `${playerBaseUrl}${this.videoId}`;
+      },
+    },
+
+    // computed: {},
+    watch: {
+      /**
+       * This should check whether responsive or height + width are specified
+       */
+      $props: {
+        immediate: true,
+        handler() {
+          if (this.responsive && (this.width || this.height)) {
+            console.error('Either use responsive or height and width');
+          } else if (
+            (this.width && !this.height)
+            || (!this.width && this.height)
+          ) {
+            console.error('width and height both need to be specified');
+          }
+        },
+      },
+      videoUrl: {
```

Comment:
todo: I don't think we should use a watcher for this. Since the `mapSrc` computed is executed anyway and execute this test before doing anything else, this watcher should not be needed.